types/react-native/index.d.ts

```diff
@@ -8489,19 +8489,28 @@ export namespace Animated {
     export type ComponentProps<T> = T extends React.ComponentType<infer P> | React.Component<infer P> ? P : never;
 
     export interface WithAnimatedValue<T>
-      extends ThisType<
-        T extends object
-          ? { [K in keyof T]?: WithAnimatedValue<T[K]> }
-          : T extends (infer P)[]
-          ? WithAnimatedValue<P>[]
-          : T | Value | AnimatedInterpolation
-        > {}
-
-    export type AnimatedProps<T> = { [key in keyof T]: WithAnimatedValue<T[key]> };
-
-    export interface AnimatedComponent<T extends React.ComponentType<ComponentProps<T>> | React.Component<ComponentProps<T>>> extends React.FC<AnimatedProps<ComponentProps<T>>> {
-        getNode: () => T;
-    }
+    extends ThisType<
+      T extends object
+        ? { [K in keyof Omit<T, 'ref'>]?: WithAnimatedValue<T[K]> }
+        : T extends (infer P)[]
+        ? WithAnimatedValue<P>[]
+        : T | Value | AnimatedInterpolation
+    > {}
+
+  export type AnimatedProps<T> = { [key in keyof T]: WithAnimatedValue<T[key]> };
+
+  export interface AnimatedComponent<
+    T extends React.ComponentType<ComponentProps<T>> | React.Component<ComponentProps<T>>
+  >
+    extends React.FC<
+      AnimatedProps<ComponentProps<T>> & {
+        ref?:
+          | React.RefObject<AnimatedComponent<T> | undefined>
```

Comment:
Does this also include the ref callback form? `ref={(ref) => ...}`

Reply:
No, That's my mistake. please show next commit.